src/token.rs

```diff
@@ -453,12 +391,9 @@ impl Token {
 
     fn get_so_login_data(&mut self) -> KResult<()> {
         if self.so_login.pin.is_none() {
-            let obj = match self.objects.get(&"0".to_string()) {
-                Some(o) => o,
-                None => return err_rv!(CKR_GENERAL_ERROR),
-            };
-            let (pin, max) =
-                self.validate_pin_obj(obj, "SO PIN".to_string())?;
+            let uid = "0".to_string();
+            let obj = self.storage.fetch_by_uid(&uid)?;
+            let (pin, max) = Self::validate_pin_obj(obj, "SO PIN".to_string())?;
```

Comment:
Well the function validates the pin object, not the pin, so that's where the name was from.
But you have a point, I reworked this area together with masking the PIN objects to be more straightforward.